google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/GapicSpannerRpc.java

```diff
@@ -338,7 +339,20 @@ public GapicSpannerRpc(final SpannerOptions options) {
                             options.getInterceptorProvider(),
                             SpannerInterceptorProvider.createDefault()))
                     .withEncoding(compressorName))
-            .setHeaderProvider(mergedHeaderProvider);
+            .setHeaderProvider(mergedHeaderProvider)
+
+            // Inject client library version to `user-agent`
+            .setHeaderProvider(
```

Comment:
We are overriding the header provider here, instead of adding to it. We have to manually construct a new provider that will encapsulate the existing one the the new one:

```java
private static final String USER_AGENT_KEY = "user-agent";
private static final String CLIENT_LIBRARY_LANGUAGE = "spanner-java";

Map<String, String> headersWithUserAgent = ImmutableMap.<String, String>builder()
    .put(USER_AGENT_KEY, CLIENT_LIBRARY_LANGUAGE + "/" + GaxProperties.getLibraryVersion(GapicSpannerRpc.class)))
    .putAll(mergedHeaderProvider.getHeaders())
    .build();
final HeaderProvider headerProviderWithUserAgent = FixedHeaderProvider.create(headersWithUserAgent);

...
.setHeaderProvider(headerProviderWithUserAgent);
```

Reply:
Fixed. Thanks!